backend/backend/authentication.py

```
@@ -0,0 +1,172 @@
+        auth_header = request.META.get("HTTP_AUTHORIZATION")
+        if not auth_header:
+            return None
+
+        token_parts = auth_header.split()
+        if len(token_parts) != 2 or token_parts[0].lower() != "bearer":
+            logger.warning("Invalid Authorization header format.")
+            return None # Invalid format
+
+        return token_parts[1]
+
+    def _verify_firebase_token(self, id_token):
```

Ruff (0.8.2)

Missing return type annotation for private function `_verify_firebase_token`

(ANN202)